SelfAndPeerEvaluations/utils.py

```diff
@@ -0,0 +1,439 @@
+            average_score = round(
+                statistics.fmean(json_file[self.id][qualitative_subject]), 2
+            )
+            qualitative_difference = self_score - partner_score
+            qualitative_differences.append(qualitative_difference)
+            row.append(
+                f"Self: {self_score}\nPartner: {partner_score}\nAverage: {average_score}\nDiff (self - partner): {qualitative_difference}"
+            )
+            self.scores[qualitative_subject + "Self"].append(self_score)
+            self.scores[qualitative_subject + "Partner"].append(partner_score)
+            self.scores[qualitative_subject + "Average"].append(average_score)
+            self.scores[qualitative_subject + "Diff"].append(qualitative_difference)
+        row.append(
+            round(statistics.fmean(qualitative_differences), 2)
+        )  # average qualitative difference
+        self.scores["Qualitative Difference"].append(
+            round(statistics.fmean(qualitative_differences), 2)
+        )
+        self_score = json_file[self.id]["Project Contribution"][0]
+        partner_score = json_file[self.id]["Project Contribution"][1]
+        average_score = round(
+            statistics.fmean(json_file[self.id]["Project Contribution"]),
+            2,
+        )
+        contrib_diff = self_score - partner_score
+        row.append(
+            f"Self: {self_score}\nPartner: {partner_score}\nAverage: {average_score}\nDiff (self - partner): {contrib_diff}"
+        )
+        self.scores["Project Contribution" + "Self"].append(self_score)
+        self.scores["Project Contribution" + "Partner"].append(partner_score)
+        self.scores["Project Contribution" + "Average"].append(average_score)
+        self.scores["Project Contribution" + "Diff"].append(contrib_diff)
+
+    def __read_files(self, files: list) -> None:
+        """
+        parse through all the quiz reports and writes info about
+        qualitative_subjects to self.qualitative_subjects
+        """
+        detailed_scores_header = []
+        for file in files:
+            with open(file, "r") as f:
+                json_file = json.load(f)
+                if len(detailed_scores_header) == 0:  # only make the header once
+                    detailed_scores_header = self.__make_detailed_header(json_file)
+                    self.detailed_scores.append(detailed_scores_header)
+                if self.qualitative_subjects is None:  # only get once
+                    self.qualitative_subjects = json_file["info"][
+                        "qualitative_subjects"
+                    ]
+                self.__parse_quiz_grade(json_file)
+
+    def __write_final_score(self) -> list:
+        self.final_score = 0.4 * self.qualitative_score + 0.6 * self.contribution_score
+        return [
+            "Final Score\n(40% Qualitative + 60% Project Contribution)",
+            round(self.final_score, 2),
+        ]
```

Comment:
Would it be possible to move the weights of the qualitative and quantitative scores to either arguments or, preferably, to the json file?